### ISSUE TITLE
Did not work on chrome v49, adding logs

### ISSUE THREAD

**Issue by vvasuki** [Jagalite/Save-to-Google-Photos#1]
Chrome Version 49.0.2623.75 (64-bit)

Logs:

```
extensions::lastError:133 Unchecked runtime.lastError while running storage.set: QUOTA_BYTES_PER_ITEM quota exceeded
    at model_setStorageObject (chrome-extension://aoanickgmocpddnppajakfcafbkncdab/javascripts/Model.js:31:29)
    at model_mergeStorageObjects (chrome-extension://aoanickgmocpddnppajakfcafbkncdab/javascripts/Model.js:20:9)
    at Object.callback (chrome-extension://aoanickgmocpddnppajakfcafbkncdab/javascripts/Google.js:19:22)reportIfUnchecked @ extensions::lastError:133
extensions::uncaught_exception_handler:8 Error in response to storage.get: TypeError: Cannot read property 'length' of undefined
    at Object.callback (chrome-extension://aoanickgmocpddnppajakfcafbkncdab/javascripts/Controller.js:42:47)
    at model_getStorageObjects (chrome-extension://aoanickgmocpddnppajakfcafbkncdab/javascripts/Model.js:3:29)
    at chrome-extension://aoanickgmocpddnppajakfcafbkncdab/javascripts/Controller.js:39:13handler @ extensions::uncaught_exception_handler:8
```

**Comment by Jagalite** [comment]
interesting, not quite convinced it is a chrome v49 problem, possibly a local storage problem?

>  QUOTA_BYTES_PER_ITEM quota exceeded

**Comment by vvasuki** [comment]
I agree. There's no shortage of space on my computer - perhaps you should use a different client-side storage mechanism ([link](http://stackoverflow.com/questions/12452275/error-during-storage-set-quota-exceeded-in-chrome-extension-developing))?

**Comment by Jagalite** [comment]
I am currently storing all the albums as a key-value pair in sync storage. Probably users with a lot of albums are facing this error.

Maybe storing only active ones or splitting up the albums into multiple storages and stitch them back togeather later would work.

**Comment by vvasuki** [comment]
> Maybe storing only active ones or splitting up the albums into multiple storages and stitch them back togeather later would work.

Or you could use the indexed db storage

**Comment by Jagalite** [comment]
sounds like a good idea! will be working on integrating an indexedDB mechanism and converting all current storages.

**Comment by vvasuki** [comment]
Is this fixed?

**Comment by Jagalite** [comment]
It should be, please test

**Comment by vvasuki** [comment]
I only see the old version:
![image](https://cloud.githubusercontent.com/assets/2664797/26279388/4c288e88-3d67-11e7-9d5f-f76599d90609.png)

How to install the new one?

**Comment by vvasuki** [comment]
Albums are showing up and I can favorite a few. But, I don't seem to be able to save anything.. 

extensions::uncaught_exception_handler:8 Error in response to identity.getAuthToken: [object Object]
    at requestTokenAndSend (chrome-extension://aoanickgmocpddnppajakfcafbkncdab/javascripts/ContextMenuScript.js:27:21)
    at onContextClick (chrome-extension://aoanickgmocpddnppajakfcafbkncdab/javascripts/ContextMenuScript.js:18:9)handler @ extensions::uncaught_exception_handler:8

**Comment by vvasuki** [comment]
Also, there is a bit of a lag before the albums show up - people might mistake that for a failure. Good idea to show some reassuring message like: "please wait..."

**Comment by Jagalite** [comment]
have you tried clearing your cache/app data? https://support.google.com/accounts/answer/32050

Also, good idea about the please wait message.

**Comment by vvasuki** [comment]
Just did - I still get the same error.

**Comment by Jagalite** [comment]
Authentication error, can you try a different account?

**Comment by vvasuki** [comment]
Just did - same error.

**Comment by Jagalite** [comment]
Will have to investigate, looks like it could be something with the chrome API for authentication tokens.

**Comment by vvasuki** [comment]
> Will have to investigate, looks like it could be something with the chrome API for authentication tokens.

Strange thing is that I am able to see and select the albums (meaning that my account was successfully accessed at that stage). But not later when saving photos to google.

**Comment by Jagalite** [comment]
> Strange thing is that I am able to see and select the albums
Getting the albums seems to not require a token explicitly from the API. It is a simple GET request to this URL: GET https://picasaweb.google.com/data/feed/api/user/userID

When we try to save an image, the extension requests a token through the API and attaches it to the POST message to google servers.


can you go to: chrome://identity-internals/ - address bar

You should see something like this, if you don't, try saving any photo and the extension should try to request a token.

![token](https://cloud.githubusercontent.com/assets/8354941/26361424/ef58f1d6-3fa8-11e7-87ed-d32db51e789c.png)

**Comment by vvasuki** [comment]
>  GET https://picasaweb.google.com/data/feed/api/user/userID

Should we be using google photos api instead?

> can you go to: chrome://identity-internals/ - address bar

I only see tokens for google calendar and google play. 

> You should see something like this, if you don't, try saving any photo and the extension should try to request a token.

The extension isn't asking me anything..

**Comment by Jagalite** [comment]
> Should we be using google photos api instead?

Google doesn't have a javascript implemented client api yet, so, for now, we are stuck with creating our own calls. https://developers.google.com/picasa-web/code

> The extension isn't asking me anything..

Well, the extension doesn't ask anything of the user for authentication purposes(except when they first install). If the extension does successfully get a token it would show up in chrome://identity-internals/. 

I am sitll investigating this issue. Maybe I can send you one of my test accounts? Or you can try changing your password, that seems to invalidate any current tokens and force new tokens to be given out.

**Comment by vvasuki** [comment]
Here is what I get now with version 8.3:
'''
extensions::uncaught_exception_handler:8 Error in response to identity.getAuthToken: [object Object]
    at requestTokenAndSend (chrome-extension://aoanickgmocpddnppajakfcafbkncdab/javascripts/ContextMenuScript.js:23:25)
    at onContextClick (chrome-extension://aoanickgmocpddnppajakfcafbkncdab/javascripts/ContextMenuScript.js:17:5)
'''

**Comment by vvasuki** [comment]
I now see this token too in chrome://identity-internals/ :

```
Access Token	
Extension Name	Save to Google Photos
Extension Id	aoanickgmocpddnppajakfcafbkncdab
Token Status	Not Found
Expiration Time	Tuesday, May 23, 2017 at 11:09:22 PM
Scopes	https://picasaweb.google.com/data/
```

**Comment by Jagalite** [comment]
Ok that's good, now I had that status "not found" too, I tried saving a image again and it changed to "present"

Can you try saving an image 2 or 3 times and see if the token's status changes?

**Comment by vvasuki** [comment]
I tried saving an image 5 times, and no luck. Same token status

**Comment by Jagalite** [comment]
Might be an account related issue, changing your password might revolk all tokens and that might fix it

Additionally, maybe you can try one of my test accounts to rule out OS or chrome version issues?

**Comment by vvasuki** [comment]
I'll gladly try one of your test accounts - if it then proves to be an account problem, I'll change my password.

**Comment by vvasuki** [comment]
![image](https://cloud.githubusercontent.com/assets/2664797/26425114/8d45e1d0-4088-11e7-9c1c-d151057d0482.png)

**Comment by Jagalite** [comment]
Ok will send in an hour or two

**Comment by Jagalite** [comment]
Just tried the newest version on my account. I got the same error like you did. I logged the error in more detail and got the message "OAuth2 not granted or revoked."

Turns out it was just a flag that needed to be set to true when getting the token.

Updating now the extension now in the web store.
You might have to uninstall and reinstall the extension.
After that when you try saving an image, you will be greeted with a google login page.
